src/nitclk/templates/session.py.mako

```diff
@@ -0,0 +1,217 @@
+# This file was generated
+<%
+    import build.helper as helper
+
+    config            = template_parameters['metadata'].config
+    attributes        = helper.filter_codegen_attributes(config['attributes'])
+    functions         = config['functions']
+
+    module_name       = config['module_name']
+    c_function_prefix = config['c_function_prefix']
+
+    functions = helper.filter_codegen_functions(functions)
+%>\
+
+import array
+import ctypes
+import datetime
+import threading
+
+% if attributes:
+import ${module_name}._attributes as _attributes
+% endif
+import ${module_name}._converters as _converters
+import ${module_name}._library_singleton as _library_singleton
+import ${module_name}._visatype as _visatype
+import ${module_name}.errors as errors
+
+# Used for __repr__ and __str__
+import pprint
+pp = pprint.PrettyPrinter(indent=4)
+
+_session_instance = None
+_session_instance_lock = threading.Lock()
+
+
+# Helper functions for creating ctypes needed for calling into the driver DLL
```

Comment:
We should move to a separate file for DRY. This can be own PR.

Reply:
Yes, it should be separate since it will touch many different places.